### PR TITLE
GetExecution performance improvements

### PR DESCRIPTION
**Pull request by katrogan**
Signed-off-by: Katrina Rogan <katroganGH@gmail.com>

# TL;DR
See associated issue for more background: https://github.com/flyteorg/flyte/issues/879

This change removes all ordering in point get queries by replacing gorm First with [Take](https://gorm.io/docs/query.html#Retrieving-a-single-object). 

Also, removes the GetById call for relaunched workflow executions which was reconstructing the parent workflow execution identifier _every time_ we called get on the relaunched exec. Instead we just serialize that value once when we actually process the request to relaunch the execution.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
see above

## Tracking Issue
https://github.com/flyteorg/flyte/issues/879

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by kumare3** [comment]
There are a lot of places where we use `First`, should we be using `Take` in all of these places
```
pkg/repositories/gormimpl/execution_repo.go:    }).First(&execution)
pkg/repositories/gormimpl/execution_repo.go:    }).First(&execution)
pkg/repositories/gormimpl/launch_plan_repo.go:  }).First(&launchPlan)
pkg/repositories/gormimpl/named_entity_repo.go: tx = tx.Select(getSelectForNamedEntity(tableName, input.ResourceType)).First(&namedEntity)
pkg/repositories/gormimpl/node_execution_repo.go:       }).Preload("ChildNodeExecutions").First(&nodeExecution)
pkg/repositories/gormimpl/project_repo.go:      }).First(&project)
pkg/repositories/gormimpl/resource_repo.go:     tx.Order(priorityDescending).First(&resources)
pkg/repositories/gormimpl/resource_repo.go:     }).First(&model)
pkg/repositories/gormimpl/task_execution_repo.go:       }).Preload("ChildNodeExecution").First(&taskExecution)
pkg/repositories/gormimpl/task_repo.go: }).First(&task)
pkg/repositories/gormimpl/workflow_repo.go:     }).First(&workflow)
```

**Review by wild-endeavor — APPROVED** [review]

**Comment by wild-endeavor** [comment]
modulo ketan's comment.

**Comment by katrogan** [comment]
PR already removes Take where appropriate
